packages/components/src/tree-grid/index.js

```diff
@@ -149,7 +149,10 @@ function TreeGrid(
 					return;
```

Comment:
One of the things that seems to be missing is that this code would previously return early if the right-most cell is focused and the user presses the right key. That stopped the focus from moving when the user reached the edge of the grid.

The change to the `if` statement on line 95 now means the early return doesn't happen when at the right edge.